tarpc/src/transport/tcp.rs

```diff
@@ -0,0 +1,62 @@
+use std::io;
+use std::net::{SocketAddr, TcpListener, TcpStream, ToSocketAddrs};
+use std::time::Duration;
+
+/// A transport for TCP.
+pub struct TcpTransport<A: ToSocketAddrs>(pub A);
+impl<A: ToSocketAddrs> super::Transport for TcpTransport<A> {
+    type Listener = TcpListener;
+    fn bind(&self) -> io::Result<TcpListener> {
+        TcpListener::bind(&self.0)
+    }
+}
+
+impl<A: ToSocketAddrs> super::Transport for A {
+    type Listener = TcpListener;
+    fn bind(&self) -> io::Result<TcpListener> {
+        TcpListener::bind(self)
+    }
+}
+
+impl super::Listener for TcpListener {
+    type Dialer = TcpDialer<SocketAddr>;
+    type Stream = TcpStream;
+    fn accept(&self) -> io::Result<TcpStream> {
+        self.accept().map(|(stream, _)| stream)
```

Comment:
The problem is that unix_socket's SocketAddr might be abstract, in which case it doesn't have a path, and thus can't be used as a `Dialer` because `UnixStream::connect` requires a path.